lib/sqewer/submitter.rb

```diff
@@ -21,11 +24,26 @@ def submit!(job, **kwargs_for_send)
     else
       serializer.serialize(job)
     end
+    validate_message_for_size!(message_body, job)
+
     connection.send_message(message_body, **kwargs_for_send)
   end
-  
+
   private
-  
+
+  def validate_job_responds_to_run!(job)
+    return if job.respond_to?(:run)
+    error_message = "Submitted object is not a valid job (does not respond to #run): #{job.inspect}"
+    raise NotSqewerJob.new(error_message)
+  end
+
+  def validate_message_for_size!(message_body, job)
+    actual_bytesize = message_body.bytesize
+    return if actual_bytesize <= MAX_PERMITTED_MESSAGE_SIZE_BYTES
+    error_message = "Job #{job.class.inspect} serialized to a message which was too large (#{actual_bytesize} bytes)"
```

Comment:
Can't we use `job.inspect` here, maybe with truncation of the arguments? The error would be more useful than just seeing a class name.